src/julia-parser.scm

```diff
@@ -2664,6 +2664,17 @@
            ;; string interpolation
            (eq? (car e) 'string))))
 
+(define (parse-public s production)
+  (if (eq? (peek-token s) 'public)
+      (let ((spc (ts:space? s)))
+        (take-token s)
+        (if (memv (peek-token s) '(#\( = #\[))
+            (begin (depwarn ...)
```

Comment:
This `(depwarn ...)` is incomplete? I guess the parser warning machinery needs to be brought back in principle?  Although how much do we care about this nicety for the old parser?

(Side note - there's no sane way to do warnings from `Meta.parse()` right now, so JuliaSyntax also won't emit these in a visible way if used via `Meta.parse()` :-/)

Reply:
Yeah I wasn't sure whether to bring back all the parser depwarn code. This can be left as a todo.